src/EventStore.LogV3/Raw.cs

```diff
@@ -66,6 +69,48 @@ public struct EpochHeader {
 			}
 		}
 
+		[StructLayout(LayoutKind.Explicit, Size = Size, Pack = 1)]
+		public struct EventHeader {
+			[FieldOffset(0)] public Guid _eventId;
+			// todo: not used yet
+			[FieldOffset(16)] public int _eventTypeNumber;
+			[FieldOffset(20)] public PrepareFlags _flags;
+			[FieldOffset(24)] public int _eventSize;
+			[FieldOffset(28)] public int _systemMetadataSize;
+			[FieldOffset(32)] public int _dataSize;
```

Comment:
does the metadata size need to be in the event header too?

Reply:
at the moment its implicit from the others (since we know the overall _eventSize and know the sizes of the other two dynamically sized components (data and system meta)

but its possible we'll need it later depending on how we go about doing alignment/padding

Reply:
@timothycoleman thanks, sounds good :+1: